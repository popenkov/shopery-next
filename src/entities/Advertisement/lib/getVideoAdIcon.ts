import { LeafIcon, PriceTagIcon } from '@/shared/ui/icons';

const icons = {
  PriceTagIcon: PriceTagIcon,
  LeafIcon: LeafIcon,
};

export const getVideoAdIcon = (iconName: keyof typeof icons) => {
  return icons[iconName];
};
