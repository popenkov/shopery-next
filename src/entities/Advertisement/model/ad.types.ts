type TAdIcons = 'icon__price_tag' | 'icon__leaf';

export type TAdvertisement = {
    icon: TAdIcons;
    title: string;
    text: string;
};
