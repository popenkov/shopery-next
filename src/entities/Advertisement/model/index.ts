export { type TAdvertisement } from './ad.types';
