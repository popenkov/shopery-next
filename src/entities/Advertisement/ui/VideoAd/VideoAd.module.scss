@import '@/shared/styles/breakpoints';
@import '@/shared/styles/mixins/flexbox';
@import '@/shared/styles/mixins/box';

.videoAd {
  @include flex-column;

  gap: 20px;
  width: 100%;
}

.videoAdContent {
  position: relative;
  max-width: 536px;
  height: 100%;
  max-height: 300px;
  aspect-ratio: 50 / 28;

  @media screen and (max-width: $df-screen-mablet-content) {
    max-width: unset;
    max-height: unset;
  }
}

.videoAdAdvantages {
  @include flex-center-vertical;

  gap: 27px;
  padding: 24px 20px;
  border: 1px solid var(--monochrome-100);
  border-radius: 6px;
}

.videoAdAdvantage {
  @include flex-center-vertical;

  gap: 12px;
}

.videoAdAdvantageIcon {
  @include box(32px);

  color: #00b307;
}

.videoAdAdvantageText {
  @include flex-column;

  gap: 6px;
}

.videoAdAdvantageSubtitle {
  margin: 0;
  color: var(--monochrome-500);
  font-weight: 400;
  font-size: 13px;
  line-height: 150%;
  letter-spacing: 0;
}
