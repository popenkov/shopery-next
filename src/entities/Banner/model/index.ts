export { type TBanner } from './banner.types';
