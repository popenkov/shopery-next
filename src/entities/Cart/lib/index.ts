export { changeItemsAmountInCart, getTotalPrice, getTotalAmount } from './helpers';
