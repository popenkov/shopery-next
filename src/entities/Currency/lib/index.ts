export * from './constants';
export { getUserCurrency, setUserCurrency } from './setCurrency';
