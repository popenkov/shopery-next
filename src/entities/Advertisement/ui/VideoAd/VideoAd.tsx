import { FC, ReactNode } from 'react';
import cn from 'classnames';

import { Text } from 'shared/ui/Text';

import { getVideoAdIcon } from '../../lib';
import { TAdvertisement } from '../../model';

import cls from './VideoAd.module.scss';

type Props = {
  adData: TAdvertisement[];
  video: ReactNode;
  className?: string;
};

export const VideoAd: FC<Props> = ({ adData, video, className }) => {
  return (
    <div className={cn(cls.videoAd, className)}>
      <div className={cls.videoAdContent}>{video}</div>
      <div className={cls.videoAdAdvantages}>
        {adData.map((ad) => {
          const { icon, title, text } = ad;
          const Icon = getVideoAdIcon(icon);
          return (
            <div className={cls.videoAdAdvantage} key={ad.title}>
              <Icon className={cls.videoAdAdvantageIcon} />
              <div className={cls.videoAdAdvantageText}>
                <Text variant="body_s" weight="medium" as="h3">
                  {title}
                </Text>
                <p className={cls.videoAdAdvantageSubtitle}>{text}</p>
              </div>
            </div>
          );
        })}
      </div>
    </div>
  );
};
