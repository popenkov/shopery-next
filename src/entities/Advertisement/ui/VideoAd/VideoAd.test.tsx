import React from 'react';
import { render } from '@testing-library/react';

import * as methods from '../../lib';
import { TAdvertisement } from '../../model';

import { VideoAd } from './VideoAd';

const adData: TAdvertisement[] = [
  {
    icon: 'PriceTagIcon',
    title: 'Title 1',
    text: 'Text 1',
  },
  {
    icon: 'LeafIcon',
    title: 'Title 2',
    text: 'Text 2',
  },
];

const video = <div>Video content</div>;

describe('VideoAd component', () => {
  it('renders correctly', () => {
    const { container } = render(<VideoAd adData={adData} video={video} />);
    expect(container).toMatchSnapshot();
  });

  it('renders video content', () => {
    const { getByText } = render(<VideoAd adData={adData} video={video} />);
    expect(getByText('Video content')).toBeInTheDocument();
  });

  it('renders ad advantages', () => {
    const { getAllByTestId } = render(<VideoAd adData={adData} video={video} />);
    const advantages = getAllByTestId('advantage');
    expect(advantages).toHaveLength(2);
  });

  it('renders ad advantage icons', () => {
    const { container } = render(<VideoAd adData={adData} video={video} />);
    const icons = container.querySelectorAll('svg');
    expect(icons).toHaveLength(2);
  });

  it('renders ad advantage titles and subtitles', () => {
    const { getAllByText } = render(<VideoAd adData={adData} video={video} />);
    const titles = getAllByText(/Title [1-2]/);
    const subtitles = getAllByText(/Text [1-2]/);
    expect(titles).toHaveLength(2);
    expect(subtitles).toHaveLength(2);
  });

  it.skip('calls getVideoAdIcon with correct icon', () => {
    const getVideoAdIconSpy = jest.spyOn(methods, 'getVideoAdIcon');
    render(<VideoAd adData={adData} video={video} />);
    expect(getVideoAdIconSpy).toHaveBeenCalledTimes(2);
    expect(getVideoAdIconSpy).toHaveBeenCalledWith('PriceTagIcon');
    expect(getVideoAdIconSpy).toHaveBeenCalledWith('LeafIcon');
  });
});
