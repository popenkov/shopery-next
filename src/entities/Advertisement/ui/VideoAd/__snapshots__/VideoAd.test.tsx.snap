// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoAd component renders correctly 1`] = `
<div>
  <div
    class="videoAd"
  >
    <div
      class="videoAdContent"
    >
      <div>
        Video content
      </div>
    </div>
    <div
      class="videoAdAdvantages"
    >
      <div
        class="videoAdAdvantage"
        data-testid="advantage"
      >
        <svg
          class="videoAdAdvantageIcon"
          data-testid="test-icon"
          fill="none"
          height="33"
          viewBox="0 0 32 33"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_460_43081)"
          >
            <path
              d="M28.0671 24.7501C28.326 24.7501 28.5359 24.5402 28.5359 24.2813V13.0313C28.5359 12.4718 28.3178 11.8995 27.9219 11.4198L24.4454 7.20769C24.2633 6.987 24.0539 6.801 23.8265 6.65287C23.8405 6.50962 23.8484 6.36494 23.8484 6.21887V0.968872C23.8484 0.709997 23.6385 0.500122 23.3797 0.500122C23.1208 0.500122 22.9109 0.709997 22.9109 0.968872C22.9109 0.968872 22.91 6.26356 22.9096 6.28593C22.7762 6.26231 22.6409 6.25012 22.5047 6.25012H16.4422C16.0154 6.25012 15.597 6.36987 15.2224 6.58994C15.2062 6.59575 15.1902 6.60225 15.1744 6.60994L10.605 8.83568C10.0987 9.08962 9.69392 9.51556 9.46498 10.035C9.46248 10.0407 9.46011 10.0464 9.45786 10.0521L7.56861 14.8441C7.47367 15.085 7.59192 15.3572 7.83279 15.4522C7.88923 15.4744 7.94736 15.4849 8.00461 15.4849C8.19148 15.4849 8.36811 15.3724 8.44079 15.188L10.3264 10.4052C10.4682 10.0884 10.7161 9.82887 11.0204 9.67612L13.4347 8.50019L11.0249 11.4199C10.6641 11.857 10.4734 12.4143 10.4734 13.0314V29.2189C10.4734 29.4074 10.49 29.592 10.52 29.7719L5.68992 27.6316C4.59879 27.1482 4.10429 25.864 4.58779 24.769C4.58986 24.7642 4.59186 24.7594 4.59386 24.7547L7.64904 17.1619C7.74567 16.9217 7.62936 16.6487 7.38917 16.5521C7.14886 16.4554 6.87598 16.5718 6.77929 16.8119L3.72679 24.3976C3.04042 25.9632 3.74942 27.7971 5.31011 28.4886L10.9741 30.9984C10.981 31.0014 10.9881 31.0037 10.9951 31.0064C11.57 31.9046 12.5639 32.5001 13.6922 32.5001H25.2547C27.0333 32.5001 28.5359 30.9974 28.5359 29.2188V26.4688C28.5359 26.2099 28.326 26.0001 28.0672 26.0001C27.8083 26.0001 27.5984 26.2099 27.5984 26.4688V29.2188C27.5984 30.4893 26.5251 31.5626 25.2547 31.5626H13.6922C12.4343 31.5626 11.4109 30.5112 11.4109 29.2188V13.0313C11.4109 12.6289 11.5243 12.2875 11.748 12.0166L15.2244 7.80444C15.548 7.41244 15.9919 7.18756 16.4422 7.18756H22.5047C22.593 7.18756 22.681 7.1965 22.7679 7.21337C22.4745 8.21406 21.7476 9.04906 20.766 9.4675C20.5094 9.03819 20.0401 8.75006 19.5046 8.75006C18.6947 8.75006 18.0359 9.40894 18.0359 10.2188C18.0359 11.0287 18.6947 11.6876 19.5046 11.6876C20.2535 11.6876 20.8729 11.1239 20.962 10.3986C22.2244 9.92112 23.179 8.90937 23.6044 7.67444C23.6452 7.7155 23.6847 7.75869 23.7224 7.80444L27.1989 12.0166C27.4565 12.3287 27.5984 12.6891 27.5984 13.0314V24.2814C27.5984 24.5402 27.8082 24.7501 28.0671 24.7501ZM19.5046 10.7501C19.2117 10.7501 18.9734 10.5117 18.9734 10.2188C18.9734 9.92587 19.2117 9.68756 19.5046 9.68756C19.7975 9.68756 20.0359 9.92587 20.0359 10.2188C20.0359 10.5117 19.7975 10.7501 19.5046 10.7501Z"
              fill="currentColor"
            />
            <path
              d="M15.4424 25.4375C15.5623 25.4375 15.6823 25.3917 15.7738 25.3002L23.7738 17.3002C23.9569 17.1171 23.9569 16.8203 23.7738 16.6373C23.5908 16.4542 23.294 16.4542 23.111 16.6373L15.111 24.6373C14.8121 24.9159 15.0414 25.4517 15.4424 25.4375Z"
              fill="currentColor"
            />
            <path
              d="M16.4414 20.5C17.5614 20.5 18.4727 19.5887 18.4727 18.4687C18.4727 17.3487 17.5614 16.4375 16.4414 16.4375H16.3789C15.2589 16.4375 14.3477 17.3487 14.3477 18.4687C14.3477 19.5887 15.2589 20.5 16.3789 20.5H16.4414ZM15.2852 18.4687C15.2852 17.8657 15.7758 17.375 16.3789 17.375H16.4414C17.0445 17.375 17.5352 17.8657 17.5352 18.4687C17.5352 19.0718 17.0445 19.5625 16.4414 19.5625H16.3789C15.7758 19.5625 15.2852 19.0718 15.2852 18.4687Z"
              fill="currentColor"
            />
            <path
              d="M22.5039 25.4375H22.5664C23.6864 25.4375 24.5977 24.5262 24.5977 23.4062C24.5977 22.2862 23.6864 21.375 22.5664 21.375H22.5039C21.3839 21.375 20.4727 22.2862 20.4727 23.4062C20.4727 24.5262 21.3839 25.4375 22.5039 25.4375ZM22.5039 22.3125H22.5664C23.1695 22.3125 23.6602 22.8032 23.6602 23.4062C23.6602 24.0093 23.1695 24.5 22.5664 24.5H22.5039C21.9008 24.5 21.4102 24.0093 21.4102 23.4062C21.4102 22.8032 21.9008 22.3125 22.5039 22.3125Z"
              fill="currentColor"
            />
          </g>
          <defs>
            <clippath
              id="clip0_460_43081"
            >
              <rect
                fill="currentColor"
                height="32"
                transform="translate(0 0.5)"
                width="32"
              />
            </clippath>
          </defs>
        </svg>
        <div
          class="videoAdAdvantageText"
        >
          <h3
            class="text primary body_s left medium"
          >
            Title 1
          </h3>
          <p
            class="videoAdAdvantageSubtitle"
          >
            Text 1
          </p>
        </div>
      </div>
      <div
        class="videoAdAdvantage"
        data-testid="advantage"
      >
        <svg
          class="videoAdAdvantageIcon"
          data-testid="test-icon"
          fill="none"
          viewBox="0 0 32 33"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            clip-path="url(#clip0_460_43091)"
          >
            <path
              d="M31.9759 1.3712C31.9127 1.17858 31.7368 1.04539 31.5343 1.03695C30.5147 0.994517 27.7279 1.00183 23.8822 1.9852C20.3025 2.90064 16.9433 4.40533 14.1675 6.3367C13.9472 6.49002 13.893 6.79283 14.0462 7.01314C14.1994 7.23345 14.5022 7.28777 14.7226 7.13452C17.4039 5.2687 20.6546 3.81377 24.1229 2.92683C25.4189 2.59545 26.7587 2.34389 28.029 2.18627C27.5082 2.4112 26.95 2.67439 26.369 2.98289C22.4401 5.06858 20.245 7.73552 19.032 9.8792C18.8726 9.16395 18.8289 8.34189 18.8286 7.97552C18.8285 7.7072 18.611 7.48977 18.3427 7.48977C18.0743 7.48977 17.8567 7.70733 17.8567 7.9757C17.8567 8.19077 17.8739 9.98302 18.4167 11.0878C17.6686 12.6852 16.9689 14.2821 16.2918 15.8286C16.0042 16.4855 15.7211 17.132 15.4404 17.7649C15.0026 16.9405 14.4256 15.4479 14.5839 13.4956C14.6055 13.2282 14.4063 12.9937 14.1387 12.972C13.872 12.9508 13.6368 13.1496 13.6151 13.4171C13.3877 16.2219 14.503 18.1801 14.9555 18.848C14.1074 20.7223 13.271 22.4484 12.3902 23.9217C11.9367 22.895 11.2263 20.9387 11.2263 18.7604C11.2263 18.492 11.0087 18.2745 10.7404 18.2745C10.472 18.2745 10.2544 18.492 10.2544 18.7604C10.2544 21.6498 11.372 24.1069 11.7742 24.8975C10.8119 26.3353 9.78436 27.4596 8.62367 28.1438C5.08779 24.3037 4.74292 17.0988 10.9201 10.4274C11.6223 9.66902 12.3929 8.94508 13.2107 8.27577C13.4184 8.10577 13.4489 7.79964 13.2789 7.59195C13.1089 7.38427 12.8027 7.35377 12.5951 7.5237C11.7429 8.22133 10.9394 8.97614 10.207 9.76708C3.69104 16.8043 4.05817 24.4746 7.79436 28.6765C4.54073 30.8661 0.514981 31.0029 0.472856 31.0041C0.204731 31.0113 -0.00683118 31.2344 0.000168815 31.5026C0.00710632 31.7665 0.223231 31.9758 0.485731 31.9758C0.490044 31.9758 0.494419 31.9757 0.498731 31.9756C0.685231 31.9707 4.97954 31.83 8.49723 29.3732C9.56311 30.2429 11.235 30.772 13.2621 30.772C15.557 30.772 18.3072 30.0937 21.148 28.4638C23.7798 26.9538 25.6666 24.6438 26.7561 21.5981C27.6891 18.9901 28.0037 15.8521 27.666 12.5234C27.0639 6.58933 29.3056 3.91164 31.8177 1.90202C31.9759 1.77533 32.0389 1.56383 31.9759 1.3712ZM27.6832 5.34633C26.7273 7.34727 26.4054 9.72702 26.699 12.6214C27.0222 15.8071 26.7255 18.798 25.8409 21.2707C24.8315 24.0926 23.0898 26.229 20.6643 27.6208C17.6035 29.377 15.0344 29.7749 13.418 29.7995C11.7655 29.8254 10.3365 29.4726 9.37767 28.8195C10.1757 28.3018 10.9084 27.6075 11.5969 26.7693C11.6348 26.7696 11.6757 26.7698 11.7214 26.7698C12.1986 26.7698 13.0877 26.7493 14.1367 26.6268C16.2354 26.3815 17.8898 25.8636 19.0541 25.0875C19.2774 24.9386 19.3377 24.6369 19.1889 24.4136C19.04 24.1904 18.7383 24.1301 18.515 24.2789C16.7214 25.4746 13.8324 25.7343 12.3412 25.7856C13.2865 24.4374 14.1572 22.8185 15.0159 21.0255C15.8609 21.0234 18.104 20.8406 20.2367 19.1404C20.4465 18.9731 20.481 18.6673 20.3137 18.4575C20.1464 18.2476 19.8407 18.2132 19.6309 18.3805C18.0646 19.6291 16.407 19.9518 15.4841 20.0291C16.0444 18.8166 16.6045 17.5374 17.182 16.2185C17.48 15.5377 17.7827 14.8468 18.0915 14.1512C18.1248 14.152 18.16 14.1525 18.197 14.1525C18.9295 14.1525 20.3882 13.9738 21.9912 12.7861C22.2068 12.6264 22.2521 12.322 22.0923 12.1064C21.9325 11.8908 21.6282 11.8455 21.4125 12.0053C20.4235 12.738 19.5133 13.0133 18.9238 13.1152C18.7809 13.14 18.6501 13.1556 18.5319 13.1656C18.8151 12.5371 19.1042 11.9065 19.401 11.2768C21.2114 7.43527 24.423 5.12058 26.7985 3.85527C27.8283 3.30677 28.7948 2.90214 29.5879 2.61302C28.8911 3.34727 28.2111 4.24133 27.6832 5.34633Z"
              fill="currentColor"
            />
          </g>
          <defs>
            <clippath
              id="clip0_460_43091"
            >
              <rect
                fill="currentColor"
                height="32"
                transform="translate(0 0.5)"
                width="32"
              />
            </clippath>
          </defs>
        </svg>
        <div
          class="videoAdAdvantageText"
        >
          <h3
            class="text primary body_s left medium"
          >
            Title 2
          </h3>
          <p
            class="videoAdAdvantageSubtitle"
          >
            Text 2
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
